cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1874,6 +1879,32 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
             return withVoid(fut);
         }
 
+        /**
+         * Get the cluster Id of the Kafka cluster
+         * 
+         * @return
+         */
+        Future<ReconciliationState> kafkaGetClusterId() {
+            String clusterId = null;
+            log.debug("{}: Attempt to get clusterId", reconciliation);
+            List<String> podNames = IntStream.range(0, kafkaCluster.getReplicas()).boxed().collect(Collectors.toList()).stream().map(podId -> KafkaCluster.kafkaPodName(this.name, podId)).collect(Collectors.toList());
+            try {
+                String bootstrapHostnames = podNames.stream().map(podName -> KafkaCluster.podDnsName(this.namespace, this.name, podName) + ":" + KafkaCluster.REPLICATION_PORT).collect(Collectors.joining(","));
+                log.debug("{}: Creating AdminClient for clusterId using {}", reconciliation, bootstrapHostnames);
+                Admin kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostnames, this.clusterCa.caCertSecret(), this.oldCoSecret, "cluster-operator");
+                clusterId = kafkaAdmin.describeCluster().clusterId().get();
+            } catch (KafkaException e) {
+                log.debug("{}: Kafka exception getting clusterId {}", reconciliation, e.getMessage());
+            } catch (Exception e) {
+                log.debug("{}: Exception getting clusterId {}", reconciliation, e.getMessage());
+            }
```

Comment:
What are the situations when this happens? In general:
* Is it intentional to continue as success when it fails instead of failing the reconciliation?
* Should the error really be only on debug level?

Both of these IMHO suggest that this is expected to fail regularly while the failure does not indicate any problems. So I would like to understand why.

Reply:
As I said above, I didn't want to make an error here critical. Either it is transient and it will eventually work or the set up is so screwed up that something else more fundamental will blow up. But that can be changed, of course.